util/certs.go

```diff
@@ -0,0 +1,95 @@
+package util
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base32"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+)
+
+func GenerateCert(ethpriv *ecdsa.PrivateKey) (*x509.Certificate, *ecdsa.PrivateKey, error) {
+	var issuerCommonName = new(bytes.Buffer)
+	// x509 Certificate signed with an randomly generated ecdsa key
+	// Certificate contains signature of ecdsa publick with ethprivate key
+	priv, err := ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ethPubKey = btcec.PublicKey(ethpriv.PublicKey)
+	issuerCommonName.Write(ethPubKey.SerializeCompressed())
+	issuerCommonName.WriteByte('@')
+
+	serializedPubKey, err := x509.MarshalPKIXPublicKey(priv.Public())
+	if err != nil {
+		return nil, nil, err
+	}
+	// Issuer must be signed with ethkey
+	var prv = btcec.PrivateKey(*ethpriv)
+	signature, err := prv.Sign(chainhash.DoubleHashB(serializedPubKey))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to sign public key: %v", err)
+	}
+	issuerCommonName.Write(signature.Serialize())
+
+	template := &x509.Certificate{
+		SerialNumber: big.NewInt(100),
+		Subject: pkix.Name{
+			CommonName: base32.StdEncoding.EncodeToString(issuerCommonName.Bytes()),
+		},
+		NotBefore: time.Now().Add(-time.Hour * 24 * 7),
```

Comment:
Should we be able to configure this?

Reply:
Makes more sense to configure NotAfter :)